### ISSUE TITLE
Invalid serialization of reindex request when using op type

### ISSUE THREAD

**Issue by gpetrou** [elastic/elasticsearch-net#8006]
**Elastic.Clients.Elasticsearch version**: 8.11.0

**Elasticsearch version**: 8.2.2

**.NET runtime version**: net8.0

**Operating system version**: Windows 11

**Description of the problem including expected versus actual behavior**:
Trying to use `ReindexAsync` with `OpType` results in an invalid request.

**Steps to reproduce**:
```
Elastic.Clients.Elasticsearch.ReindexRequest request = new()
{
    Source = new Elastic.Clients.Elasticsearch.Core.Reindex.Source()
    {
         Index = Elastic.Clients.Elasticsearch.Indices.Parse("test-src")
    },
    Dest = new Elastic.Clients.Elasticsearch.Core.Reindex.Destination()
    {
         Index = "test-dest",
         OpType = Elastic.Clients.Elasticsearch.OpType.Create
    }
};

var response = await _elasticsearchClient.ReindexAsync(request);
```

```
{
  "dest": {
    "index": "test-dest",
    "op_type": {
      "value": "create"
    }
  },
  "source": {
    "index": "test-src"
  }
}
```

`ServerError: Type: x_content_parse_exception Reason: "[4:16] [reindex] failed to parse field [dest]" CausedBy: "Type: x_content_parse_exception Reason: "[4:16] [dest] op_type doesn't support values of type: START_OBJECT""`

**Expected behavior**
The request works.

**Provide `ConnectionSettings` (if relevant)**:

**Provide `DebugInformation` (if relevant)**:

**Comment by gpetrou** [comment]
@flobernd this is still not working in 8.13.3.

**Comment by flobernd** [comment]
@gpetrou Yes, that is expected. The manually crafted `OpType` type needs a specialized JSON converter. Thanks for re-testing with the latest version! This saves me some time. I'm soon going through all the open issues to check, which of them can be closed after the 8.13.* release.
